src/util/scheduler.hpp

```diff
@@ -77,7 +78,7 @@ class Scheduler {
 
     // Get the scheduler for frozen Realms. This scheduler does not support
     // notifications and does not perform any thread checking.
-    static std::shared_ptr<Scheduler> get_frozen();
+    static std::shared_ptr<Scheduler> get_frozen(VersionID version = VersionID());
```

Comment:
When the default value is used here repeatedly, we'll return the cached realm again even if it has been frozen on different versions (ie requesting the latest frozen Realm, writing and advancing, then requesting the latest frozen Realm). In that scenario the version stored by the frozen scheduler gets out of sync with the actual version of the Realm, so I think this solution is insufficient?

Reply:
I only added the default value because I was too lazy to go and modify all the invocations from tests. In reality, all places where `get_frozen` is called from non-test code, a non-default version is passed. In the tests, we don't appear to care about caching Realms, so passing the default version is fine.

Reply:
I don't know how SDK's are using this, but it is a public API and I'd prefer a design that doesn't give SDK's the option to make to make this mistake. If that means passing the version into `do_get_cached_realm` I'm fine with that even if it seems less elegant.

Reply:
`Scheduler::get_frozen()` could theoretically be called directly, but in practice none of the SDKs do.

Reply:
I can also remove the default argument for `get_frozen` and update the tests? This way we'll force consumers to think what version they're passing in.

Reply:
Ok sure, not blocking for me as there doesn't seem to be a much better way at this point. Bonus points if you can assert that the scheduler's frozen version matches the Realm's version somewhere.